apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx

```diff
@@ -203,7 +205,19 @@ export function ActivityDetails(props: {
   const isCustomRecipientWarningBannerVisible = !useIsReceiverWalletBannerHidden(id) && order && isPending(order)
   const hideCustomRecipientWarning = useHideReceiverWalletBanner()
   const orderProgressBarV2Props = useOrderProgressBarV2Props({ activityDerivedState, chainId })
-  const surplusData = useGetSurplusData(order)
+  const setShowProgressBar = useAddOrderToSurplusQueue() // TODO: not exactly the proper tool, rethink this
```

Comment:
I'm re-using the logic for triggering the surplus modal.
Not happy with this, but it's working.

In the good to haves to refactor it.